e2e-test/src/test/features/BigQueryToDistinct.feature

```diff
@@ -0,0 +1,99 @@
+  Scenario: To verify data is getting transferred from BigQuery to BigQuery sink plugin successfully with distinct fields
+    Given Open Datafusion Project to configure pipeline
+    When Select plugin: "BigQuery" from the plugins list as: "Source"
+    When Expand Plugin group in the LHS plugins list: "Analytics"
+    When Select plugin: "Distinct" from the plugins list as: "Analytics"
+    Then Connect plugins: "BigQuery" and "Distinct" to establish connection
+    When Expand Plugin group in the LHS plugins list: "Sink"
+    When Select plugin: "BigQuery" from the plugins list as: "Sink"
+    Then Connect plugins: "Distinct" and "BigQuery2" to establish connection
+    Then Navigate to the properties page of plugin: "BigQuery"
+    Then Replace input plugin property: "projectId" with value: "projectId"
+    Then Enter input plugin property: "datasetProjectId" with value: "projectId"
+    Then Enter input plugin property: "referenceName" with value: "BQReferenceName"
+    Then Enter input plugin property: "dataset" with value: "dataset"
+    Then Enter input plugin property: "table" with value: "bqSourceTable"
+    Then Click on the Get Schema button
+    Then Validate "BigQuery" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "Distinct"
+    Then Enter Distinct plugin Fields "distinctGCSValidFields"
+    Then Enter input plugin property: "numberOfPartitions" with value: "distinctValidPartitions"
+    Then Click on the Get Schema button
+    Then Verify the Output Schema matches the Expected Schema: "distinctOutputSchema"
+    Then Validate "Distinct" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "BigQuery2"
+    Then Replace input plugin property: "projectId" with value: "projectId"
+    Then Enter input plugin property: "datasetProjectId" with value: "projectId"
+    Then Enter input plugin property: "referenceName" with value: "BQReferenceName"
+    Then Enter input plugin property: "dataset" with value: "dataset"
+    Then Enter input plugin property: "table" with value: "bqTargetTable"
+    Then Click plugin property: "truncateTable"
+    Then Click plugin property: "updateTableSchema"
+    Then Validate "BigQuery2" plugin properties
+    Then Close the Plugin Properties page
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Close the pipeline logs
+    Then Validate OUT record count of distinct is equal to IN record count of sink
```

Comment:
Then Validate target table "bqTargetTable" records count is equal to source table "bqSourceTable" records count with distinct fields "distinctBQFieldsList"

implementation -> fetch the record count with distinct query from source table and assert with total records count of target table

Reply:
Added the validation step